sway-core/src/abi_generation/abi_str.rs

```diff
@@ -208,6 +211,19 @@ impl TypeInfo {
     }
 }
 
+fn handle_abi_name_attribute(call_path: &CallPath, attributes: &transform::Attributes) -> CallPath {
+    let mut abi_call_path = call_path.clone();
+    if let Some(abi_name_attr) = attributes.abi_name() {
+        let name = abi_name_attr.args.first().unwrap();
+        let literal = match name.value.as_ref().unwrap() {
+            Literal::String(lit_string) => lit_string,
+            _ => unreachable!(),
+        };
+        abi_call_path.suffix = BaseIdent::new(literal.span.clone().trim_quotes());
```

Comment:
A general question to renaming ABI elements. What if we rename it to something that already exist and is also serialized into ABI? Shouldn't we provide any checks here?

Reply:
I gave this a bit of a thought. Giving names is a serious business. Aside from renaming that can clash with an existing item what about invalid names? E.g.
- `name = ""`
- `name = "this !s n0t an identif1er"`
- `name = "__this_also_not"`
- `name = "this::looks::like::a::path"`
- `name = "will this \" break ABI generation or be encoded?"`

What will SDKs do in these cases? How do they handle invalid names, if at all? I wouldn't be surprised if expecting a valid name is a part of an unwritten contract for ABI consistency.

I expect strong checks here for consistency. Honestly, I would go that far and give warnings for non-idiomatic names.